src/Tools/dotnet-monitor/Extensibility/ProgramExtension.cs

```diff
@@ -65,7 +73,15 @@ public async Task<EgressArtifactResult> EgressArtifact(ExtensionEgressPayload co
             // This is really weird, yes, but this is one of 2 overloads for [Stream].WriteAsync(...) that supports a CancellationToken, so we use a ReadOnlyMemory<char> instead of a string.
             ReadOnlyMemory<char> NewLine = new ReadOnlyMemory<char>("\r\n".ToCharArray());
 
-            string programRelPath = Path.Combine(Path.GetDirectoryName(_exePath), Declaration.Program);
+            string exeName = Declaration.Program;
+
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && Path.GetExtension(exeName) == ".exe")
```

Comment:
I think I would rather have it the opposite way: extensions declare their executable without a file extension and then we tack on ".exe" for the Windows platform.